xdsl/transforms/scf_parallel_loop_tiling.py

```diff
@@ -0,0 +1,156 @@
+from dataclasses import dataclass
+
+from xdsl.dialects import affine, arith
+from xdsl.dialects.builtin import IndexType, IntegerAttr, ModuleOp
+from xdsl.dialects.scf import ParallelOp, Yield
+from xdsl.ir import Block, MLContext, Operation, Region, SSAValue
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.utils.hints import isa
+
+
+@dataclass
+class ScfParallelLoopTilingPattern(RewritePattern):
+    tile_sizes: list[int]
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ParallelOp, rewriter: PatternRewriter, /):
+        # Only tile the innermost parallel loop
+        if any(isinstance(o, ParallelOp) for o in op.body.walk()):
```

Comment:
do you need to walk the body? would this get different results?
```suggestion
        if any(isinstance(o, ParallelOp) for o in op.body.block.ops):
```

Reply:
Yep, could be a parallel nested in something else, I want to catch those